api_tests/applications/views/test_application_detail.py

```diff
@@ -1,35 +1,41 @@
+import pytest
 import mock
 
-from nose.tools import *  # flake8: noqa
-
 from website.models import ApiOAuth2Application, User
 from website.util import api_v2_url
-
 from tests.base import ApiTestCase
+from api_tests.utils import assert_dict_contains_subset
 from osf_tests.factories import ApiOAuth2ApplicationFactory, AuthUserFactory
 
 
 def _get_application_detail_route(app):
-    path = "applications/{}/".format(app.client_id)
+    path = 'applications/{}/'.format(app.client_id)
     return api_v2_url(path, base_route='/')
 
 
 def _get_application_list_url():
-    path = "applications/"
+    path = 'applications/'
     return api_v2_url(path, base_route='/')
 
+@pytest.fixture()
+def user():
+    return AuthUserFactory()
 
-class TestApplicationDetail(ApiTestCase):
-    def setUp(self):
-        super(TestApplicationDetail, self).setUp()
+@pytest.mark.django_db
+@pytest.mark.usefixtures('app')
+class TestApplicationDetail:
 
-        self.user1 = AuthUserFactory()
-        self.user2 = AuthUserFactory()
+    @pytest.fixture()
+    def user_app(self, user):
+        return ApiOAuth2ApplicationFactory(owner=user)
 
-        self.user1_app = ApiOAuth2ApplicationFactory(owner=self.user1)
-        self.user1_app_url = _get_application_detail_route(self.user1_app)
+    @pytest.fixture()
+    def user_app_url(self, user_app):
+        return _get_application_detail_route(user_app)
 
-        self.missing_id = {
+    @pytest.fixture()
+    def missing_id(self):
```

Comment:
I would still like to see https://github.com/CenterForOpenScience/osf.io/pull/7257#discussion_r118514062 implemented, unless you have any objections.